components/tikv_util/src/mpsc/batch.rs

```diff
@@ -341,19 +353,18 @@ mod tests {
 
         let msg_counter = Arc::new(AtomicUsize::new(0));
         let msg_counter1 = Arc::clone(&msg_counter);
-        let pool = CpuPool::new(1);
+        let pool = Builder::new()
+            .threaded_scheduler()
+            .core_threads(1)
+            .build()
+            .unwrap();
         let _res = pool.spawn(rx.for_each(move |_| {
             msg_counter1.fetch_add(1, Ordering::AcqRel);
-            Ok(())
+            future::ready(())
         }));
 
         // Wait until the receiver is suspended.
-        loop {
-            thread::sleep(time::Duration::from_millis(10));
-            if !tx.state.recv_task.load(Ordering::SeqCst).is_null() {
-                break;
-            }
-        }
+        thread::sleep(time::Duration::from_millis(100));
```

Comment:
I suggest to keep original loop. Running tests concurrently can slow down the test quite a lot.